src/parser/parsers/heroes.py

```diff
@@ -93,6 +109,108 @@ def _parse_hero_weapon(self, hero_value):
         weapon_stats['DPS'] = weapon_stats['BulletDamage'] * weapon_stats['RoundsPerSecond']
         return weapon_stats
 
+    # Parse the stats that are listed in the UI in game
+    def _parse_stats_ui(self, hero_value):
+        parsed_stats_ui = {}
+
+        """
+            Within m_heroStatsUI
+            Transform each value within m_vecDisplayStats array
+            
+            {
+                "m_eStatType": "EMaxHealth",
+                "m_eStatCategory": "ECitadelStat_Vitality"
+            }
+
+            to a dict entry
+            "MaxHealth": "Vitality"
+
+            
+
+            Within m_ShopStatDisplay
+            Transform
+            "m_eWeaponStatsDisplay": {
+                "m_vecDisplayStats": [
+                    "EBulletDamage",
+                    "EBaseWeaponDamageIncrease",
+                    "ERoundsPerSecond",
+                    "EFireRate",
+                    "EClipSize",
+                    "EClipSizeIncrease",
+                    "EReloadTime",
+                    "EReloadSpeed",
+                    "EBulletSpeed",
+                    "EBulletSpeedIncrease",
+                    "EBulletLifesteal"
+                ],
+                "m_vecOtherDisplayStats": [
+                    "ELightMeleeDamage",
+                    "EHeavyMeleeDamage"
+                ],
+                "m_eWeaponAttributes": "EWeaponAttribute_RapidFire | EWeaponAttribute_MediumRange"
+            },
+
+            to
+
+            "BulletDamage": "Weapon",
+            "BaseWeaponDamageIncrease": "Weapon",
+            "RoundsPerSecond": "Weapon",
+            "FireRate": "Weapon",
+            "ClipSize": "Weapon",
+            "ClipSizeIncrease": "Weapon",
+            "ReloadTime": "Weapon",
+            "ReloadSpeed": "Weapon",
+            "BulletSpeed": "Weapon",
+            "BulletSpeedIncrease": "Weapon",
+            "BulletLifesteal": "Weapon",
+            "LightMeleeDamage": "Weapon",
+            "HeavyMeleeDamage": "Weapon"
+            "WeaponAttribute_RapidFire": "Weapon"
+            "WeaponAttribute_MediumRange": "Weapon"
+            
+        """
```

Comment:
I'm not sure the logic is complex enough to warrant such a wall of text. 
Comments are fine, but place them above their relevant block of code and keep it concise

Reply:
Removed the excessive examples; its now as short as possible